packages/taquito/src/wallet/legacy.ts

```diff
@@ -32,8 +33,35 @@ export class LegacyWalletProvider implements WalletProvider {
     return attachKind(await params(), OpKind.INCREASE_PAID_STORAGE);
   }
 
+  async mapFailingNoopParamsToWalletParams(params: () => Promise<WalletFailingNoopParams>) {
+    return attachKind(await params(), OpKind.FAILING_NOOP);
+  }
+
   async sendOperations(params: WalletParamsWithKind[]) {
     const op = await this.context.batch.batch(params as any).send();
     return op.hash;
   }
+
+  async sign(signingRequest: { payload: string; signingType: 'operation' | 'micheline' | 'raw' }) {
+    let magicByte: Uint8Array | undefined;
+    switch (signingRequest.signingType) {
+      case 'micheline':
```

Comment:
slight nitpick, but we could maybe make this an enum along with 'operation' and 'raw' as SigningType?